tests/binning_test.py

```diff
@@ -897,7 +897,7 @@ def test_bin_with_explicit_lower_precision_drops_rows_outside_domain():
     x = sc.linspace('x', 0.0, 1.0, 3, unit='m', dtype='float32')
     da = table.bin(x=x)
     size = da.bins.size().sum().value
-    table.coords['x'].values[0] = 2.0 * np.finfo(np.float32).max
+    table.coords['x'].values[0] = np.float64(2.0) * np.finfo(np.float32).max
```

Comment:
Type promotion rules have changed: https://numpy.org/devdocs/numpy_2_0_migration_guide.html#changes-to-numpy-data-type-promotion